src/performance/utils.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { Profiler, useCallback, useRef } from 'react';
+
+const events = new Map<string, number>();
+
+export function measureEventStart(name: string) {
+  events.set(name, global.performance.now());
+}
+
+export function measureEventEnd(name: string) {
+  const total = global.performance.now() - events.get(name)!;
+  events.set(name, total);
+  global.console.log(`Event ${name}: ${total.toFixed(2)}ms\n`);
+}
+
+const timeoutRegistry = new Map<string, any>();
+
+function createDebounceAverage(
+  title: string,
+  phase: string,
+  delay: number = 100
+) {
+  let values: number[] = [];
+  let _timeout: NodeJS.Timeout;
+
+  return (value: number) => {
+    clearTimeout(_timeout);
+    values.push(value);
+
+    _timeout = setTimeout(() => {
+      const total = values.reduce((acc, curr) => acc + curr);
```

Comment:
```suggestion
      const total = values.reduce((a, b) => a + b);
```

If I am already making stupid comments about variable names, what do you think about this instead? IMO it looks very descriptive for reduce that sums numbers.

Reply:
Disagree on this one, since we don't have only `a` and `b`, but actually the current item and the total item, I'm following the default names for the reduce operation.

Reply:
Ok 😄